sql/ddl/create/check-and-default-value.md

```diff
@@ -42,13 +41,120 @@ aspects:
+
+```sql
+CREATE TABLE voters (
+  id INTEGER PRIMARY KEY,
+  name VARCHAR,
+  age INTEGER CHECK (AGE >= 18)
+);
+```
+
+Now, if you were to insert a record which had a value for the `age` column less that 18 you would get an error. 
+
+What if you wanted to constrain the value of two columns at the same time? This is possible by creating a table check constraint. Let's say that for some reason you also want only names that start with 'T'. In this case, your table definition would look like:
+
+```sql
+CREATE TABLE voters (
+  id INTEGER PRIMARY KEY,
+  name VARCHAR,
+  age INTEGER,
+  CONSTRAINT age_and_name_check CHECK (age >= 18 AND name LIKE 'T%')
+);
+```
+
+Now let's try inserting some values in our table:
+
+```sql
+INSERT INTO voters VALUES (1, 'Abraham Lincoln', 56);
+-- This will fail because the name starts with 'A'
+INSERT INTO voters VALUES (2, 'Titus Andronicus', 33);
+-- This will be successful
+```
+
+There are two main advantages of using table constraints over column constraints. The first advantage is that they are more powerful, allowing you to use two or more columns. The second one is that they can have their own name, in our case `age_and_name_check`, which helps with:
+- understanding system messages
+- creating constraints after the table was defined
+- altering (activating, deactivating, deleting) the constraints
+ 
+### Default Value
 
+When creating a table you can also set a default value for your columns. This value can be a constant number like 5, a string like 'not available', a system variable or a function call that determines dynamic values like the actual timestamp.
+
+The `DEFAULT` clause affects `INSERT` commands that do not specify a value for the column in question. Let's start off by defining a simple table named `dogs` that only takes the `name` and `weight` columns.
+
+```sql
+CREATE TABLE dogs (
+  name VARCHAR,
+  weight INTEGER DEFAULT 0
+);
+```
+
+Now let's see what happens when we insert the following records in the table:
+
+```sql
+INSERT INTO dogs (name) VALUES ('Oscar');
+INSERT INTO dogs (name, weight) VALUES ('Benson', 22);
+```
+
+And this is how the table would look like:
+
+| name   | weight |
+|--------|--------|
+| Oscar  | 0      |
+| Benson | 22     |
+
+Notice that if we omit the `weight` column when using an `INSERT` command, the RDBMS will store the default value, which was 0.
 
 ---
 ## Practice
 
+Complete the following table definition to check that the `year` column is higher that 1970 and the `name` column has a default value of 'unknown'.
+
+```sql
+CREATE TABLE cars (
+  id INTEGER PRIMARY KEY,
+  name VARCHAR ??? '???',
+  year INTEGER ??? (???)
+);
+```
+
+* DEFAULT
+* unknown
+* CHECK
+* year > 1970
```

Comment:
Try adding answers here that seem a bit more viable as correct.

For example:

* DEFAULT
* unknown
* CHECK
* year > 1970
* FALLBACK
* undefined
* VERIFY
* year GT THAN 1970

Reply:
yeah, I was trying to think of some but none were coming to mind, thanks for the tips

Reply:
added the suggestions